x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engine/engine_view_logic.ts

```diff
@@ -80,5 +96,22 @@ export const EngineViewLogic = kea<MakeLogicType<EngineViewValues, EngineViewAct
         return status === Status.IDLE || (!data && status === Status.LOADING);
       },
     ],
+    isLoadingEngineSchema: [
+      () => [selectors.fetchEngineSchemaApiStatus],
+      (status: EngineViewValues['fetchEngineSchemaApiStatus']) =>
+        [Status.LOADING, Status.IDLE].includes(status),
+    ],
+    schemaFields: [
+      () => [selectors.engineSchemaData],
+      (data) =>
+        Object.entries(data?.field_capabilities?.fields ?? {})
+          .map(([name]) =>
+            Object.entries({
+              field_name: name,
+              field_type: [...Object.keys(data?.field_capabilities?.fields[name])],
```

Comment:
I think it's probably OK for this PR, but looking at the Figma we don't have a comma-delimited list of field types, we simply wrap up multiple types in a friendlier error message. We might want to double check with @julianrosado on that display.

Reply:
Ultimately, as you mentioned @kderusso , this will be wrapped in some sort of error messaging. However, since we still haven't aligned on what that messaging is or how we will display it, I hesitate to start sending @saarikabhasi down any dark forest UI paths. 

Since this isn't slated for immediate release, I'd move forward with what we have and I'll update the designs based on the output of our schema page conversations... as that may necessitate a bunch of UI changes anyways!